ui/src/ducks/app/monitoring.js

```diff
@@ -49,9 +92,60 @@ export function* fetchAlerts() {
 
 export function* fetchClusterStatus() {
   const api = yield select(state => state.config.api);
-  const result = yield call(getClusterStatus, api.url_prometheus);
-  if (!result.error) {
-    yield put(setClusterStatusAction(result.data.data.result));
+
+  const apiserverQuery = 'sum(up{job="apiserver"})';
+  const kubeSchedulerQuery = 'sum(up{job="kube-scheduler"})';
+  const kubeControllerManagerQuery = 'sum(up{job="kube-controller-manager"})';
+
+  const results = yield all([
+    call(queryPrometheus, api.url_prometheus, apiserverQuery),
+    call(queryPrometheus, api.url_prometheus, kubeSchedulerQuery),
+    call(queryPrometheus, api.url_prometheus, kubeControllerManagerQuery)
+  ]);
+
+  const apiserver = results[0];
+  const apiServerValue = apiserver.data.data.result[0].value;
```

Comment:
Maybe this pattern for checking status before putting the value could be shared (you have it three times already):

```javascript
export async function queryPrometheus(api, query) {
  try {
    const result = await axios.get(api + '/api/v1/query?query=' + query);
    return { status: result.data.status, value: result.data.data.result[0].value }
  } catch (error) {
    return { error };
  }
}
```

If you need special handling of missing `.data` / or empty `.value`, you could do it in this method as well, maybe returning an error.

Reply:
good point.
=> See PR #1268